core/src/main/web/outputdisplay/bko-tabledisplay/bko-tabledisplay.js

```diff
@@ -137,12 +137,18 @@
         $scope.exportTo = function(data, format) {
           var i;
           var j;
+          var startingColumnIndex = 1;
           var order;
           var out = '';
           var eol = '\n';
           var sep = ',';
           var qot = '"';
           var fix = function(s) { return s.replace(/"/g, '""');};
+          var model = $scope.model.getCellModel();
+          var hasIndex = model.hasIndex === "true";
```

Comment:
Should I refactor it now?
It will require changes in `beaker.r`, `beaker_runtime.py`, `beaker_runtime3.py`, `bko-tabledisplay.js` and `sessionmanager.js`.